src/v2/views/MonsterCollectionOverlay/Migration.tsx

```diff
@@ -43,9 +43,10 @@ export default function Migration({
 
   const deposit = useMemo(
     () =>
-      collectionSheet?.orderedList?.find(
-        (v) => v?.level === collectionState.level
-      )?.requiredGold,
+      collectionSheet?.orderedList
+        ?.filter((v) => v?.level <= collectionState.level)
+        .map((v) => v.requiredGold)
+        .reduce((a, b) => a + b, 0),
```

Comment:
```suggestion
        .reduce(({ a: requiredGold }, { b: requiredGold }) => a + b, 0),
```

One less loop for ya. Prettier would undoubtedly complain about this, so run it before committing this.

Reply:
Though prettier doesn't let me know it, it seems very good. But I'm stuck while matching type. I'll see it more, thanks!

Reply:
I'll append a commit accordingly, thanks for your contribution.